main.go

```diff
@@ -93,6 +98,10 @@ func main() {
 		log.Fatalf("failed to parse yaml file due to error: %s", err)
 	}
 
+	if opts.Clean {
+		cleanDestinations(getUniqueDestinations(config))
```

Comment:
any reasons we can't just combine the two functions into a single function `cleanDestinations(config)`?

Reply:
Its a matter of taste. Personally, I prefer to have more smaller functions each for its own logical task. Also, might have been helpful in project plans to separate functionality. Nonetheless, it's not anyhow critical atm, and none knows future.